File: Dagger2Ex/app/src/main/java/expencive/vk/com/dagger2ex/dagger/CarComponent.java
package expencive.vk.com.dagger2ex.dagger;

import dagger.Component;
import expencive.vk.com.dagger2ex.MainActivity;
import expencive.vk.com.dagger2ex.car.Car;

@Component(modules = {WheelsModule.class, DieselEngineModule.class})
public interface CarComponent {

    Car getCar();

    void inject(MainActivity mainActivity);
}

File: DaggerKotlin/app/src/main/java/vk/expencive/daggerkotlin/dagger/CarComponent.kt
package vk.expencive.daggerkotlin.dagger

import dagger.Component
import vk.expencive.daggerkotlin.MainActivity
import vk.expencive.daggerkotlin.car.Car

@Component(modules = [WheelsModule::class, DieselEngineModule::class])
interface CarComponent {

    fun getCar(): Car

    fun ingect(mainActivity: MainActivity)


}

